pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -147,48 +147,65 @@ func BuildSidecarVirtualHostWrapper(routeCache *Cache, node *model.Proxy, push *
 // plain non-registry hostnames
 func separateVSHostsAndServices(virtualService config.Config,
 	serviceRegistry map[host.Name]*model.Service,
-	mostSpecificWildcardIndex map[host.Name]types.NamespacedName,
+	mostSpecificWildcardVsIndex map[host.Name]types.NamespacedName,
 ) ([]string, []*model.Service) {
 	// TODO: A further optimization would be to completely rely on the index and not do the loop below
 	// However, that requires assuming that serviceRegistry never got filtered after the
 	// egressListener was created.
 	rule := virtualService.Spec.(*networking.VirtualService)
-	hosts := make([]string, 0)
+	// Stores VS hosts that don't correspond to services in the registry
+	// Currently, the only use for this list is to enable VirtualService configuration to affect
+	// traffic to hosts outside of the service registry (e.g. google.com) on port 80
+	nonServiceHosts := make([]string, 0)
```

Comment:
nit: Since you are changing these vars, may be it is better to be explicit and call
nonServiceHosts as `nonRegistryHosts` or `externalHosts` and `servicesInVirtualService` as `matchingRegistryServices` (InVirtualService is actually redundant IMO)